devman.go

```diff
@@ -0,0 +1,170 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"sync"
+
+	"github.com/ammesonb/dispersed-backup/device"
+	"github.com/ammesonb/dispersed-backup/mydb"
+)
+
+var getDevices = mydb.GetDevices
+var makeDevice = device.MakeDevice
+var addDBDevice = mydb.AddDevice
+
+// DevCommandAddDevice instructs the manager to add a new device by mount
+var DevCommandAddDevice int = 1
+
+// DevCommandReserveSpace instructs the manager to reserve an amount of space, optionally on a specific mount
+var DevCommandReserveSpace int = 2
+
+// DevCommandFreeSpace instructs the manager to free an amount of space on a given mount
+var DevCommandFreeSpace int = 3
+
+// DeviceCommand contains information needed to execute a command
+type DeviceCommand struct {
+	// Command integer, see variables above
+	command int
+	// Mountpoint and serial, for adding a new device
+	// Mountpoint may also be used to request storing a file on a specific mountpoint
+	mountPoint string
+	serial     string
+	// Space to allocate or free
+	space int64
+}
+
+// DeviceResult contains details about the executed action
+type DeviceResult struct {
+	success bool
+	message string
+	err     error
+}
+
+// DevMan contains the necessary components for interacting with the device manager goroutine
+type DevMan struct {
+	commands chan DeviceCommand
+	results  chan DeviceResult
+	lock     sync.Mutex
+}
+
+// RunManager should be used in a goroutine, and is responsible for managing available device space for file backups
+// A MutEx should be used to maintain one-to-one command -> result behavior
+func RunManager(db *sql.DB, commands <-chan DeviceCommand, results chan<- DeviceResult) {
+	devices := getDevices(db)
+
+	go func() {
+		process(&devices, db, commands, results)
+	}()
+}
+
+func process(devices *[]*device.Device, db *sql.DB, commands <-chan DeviceCommand, results chan<- DeviceResult) {
+	for command := range commands {
+		handle(command, devices, db, commands, results)
+	}
+}
+
+var handle = func(command DeviceCommand, devices *[]*device.Device, db *sql.DB, commands <-chan DeviceCommand, results chan<- DeviceResult) {
+	defer func() {
+		if r := recover(); r != nil {
+			// Ignore errors, since need to keep processing requests
+			fmt.Println("Recovered. Error:\n", r)
+			// Since only called when command received, ensure we inform the caller there was an error
+			results <- DeviceResult{false, "", fmt.Errorf("Panic during execution")}
+		}
+	}()
+
+	switch command.command {
```

Comment:
Have you considered using a jump table here? 

https://stackoverflow.com/questions/9928221/table-of-functions-vs-switch-in-golang

It's cleaner, but it may be a little slower.